catalogs/flink-iceberg-hive/flink/Dockerfile

```diff
@@ -0,0 +1,51 @@
+FROM apache/flink:1.18.1-scala_2.12-java11
+
+WORKDIR /opt/flink
+
+# Set up Hive config
+COPY conf/hive-site.xml ./conf/hive-site.xml
+# Pre-seed the SQL history because I'm nice like that
+COPY flink-sql-history /root/.flink-sql-history
+
+# Enable SQL Client to find the job manager when running it from this image
+RUN sed -i "s/jobmanager.rpc.address: localhost/jobmanager.rpc.address: flink-jobmanager/g" ./conf/flink-conf.yaml
+
+# Install required dependencies
+RUN apt-get update && \
+    apt-get install -y vim lnav
+
+RUN echo "Add Flink S3 Plugin" && \
+    mkdir ./plugins/s3-fs-hadoop && \
+    cp ./opt/flink-s3-fs-hadoop-1.18.1.jar ./plugins/s3-fs-hadoop/
+
+RUN echo "-> Install JARs: Flink's Hive connector" && \
+    mkdir -p ./lib/hive && \
+    curl https://repo1.maven.org/maven2/org/apache/flink/flink-sql-connector-hive-3.1.3_2.12/1.18.1/flink-sql-connector-hive-3.1.3_2.12-1.18.1.jar -o ./lib/hive/flink-sql-connector-hive-3.1.3_2.12-1.18.1.jar
```

Comment:
Here I'd use the `-O` option I mentioned in the blog post the other day, to keep things concise (and easier to update later on).

Reply:
Done. thanks.